### PR TITLE
Add overflow check to encoding.DecodeVarint.

### PR DESCRIPTION
**Pull request by sbunce**
After we use DecodeUvarint to decode a positive signed int we need to overflow
check to make sure the resulting uint64 doesn't overflow int64.

### CODE REVIEW

File: util/encoding/encoding_test.go [hunk 1]
```diff
@@ -362,6 +362,24 @@ func TestDecodeInvalidUvarint(t *testing.T) {
...
+	testCases := [][]byte{
+		// overflows int64.
+		{0x10, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff},
+	}
+	for _, c := range testCases {
+		func() {
+			defer func() {
+				if r := recover(); r == nil {
+					t.Error("expected panic")
+				}
```
Comment (petermattis) [inline util/encoding/encoding_test.go]: The tests which catch panics could be made mildly more robust by checking that the error message matches a regexp.

**Comment by petermattis** [comment]
LGTM

File: util/encoding/encoding_test.go (hunk 1 above)
Comment (sbunce) [inline util/encoding/encoding_test.go]: This is a good idea. I have the feeling that doing this in tests would encourage descriptive errors.

Done.

**Comment by petermattis** [comment]
Nice. LGTM.

**Comment by petermattis** [comment]
@sbunce I've added you to the cockroach organization so you can merge this when ready.

File: util/encoding/encoding_test.go [hunk 2]
```diff
@@ -346,18 +348,71 @@ func TestEncodeDecodeUvarint(t *testing.T) {
 
 // TestDecodeInvalidUvarint tests that invalid uvarint encodings panic.
 func TestDecodeInvalidUvarint(t *testing.T) {
-	testCases := [][]byte{
-		// length of 9 bytes should cause an error.
-		{0x11, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01},
+	tests := []struct{
+		name    string // name printed with errors.
```
Comment (tamird) [inline util/encoding/encoding_test.go]: nit: this struct could be extracted. :+1: otherwise

Comment (petermattis) [inline util/encoding/encoding_test.go]: You could also combine the tests by making the decode function a parameter of the struct:

```
  tests := []struct {
    ...
    decode func([]byte)
  }{
    {
      ...
      decode: func(b []byte) { DecodeUvarint(b) },
    },
  }
```

**Comment by sbunce** [comment]
Sorry I'm still learning git. My git history for this is not right.

I'm going to close this one and start over.

**Pull request closed without merging**